docs/architecture-decisions/adr013-use-node-fetch.md

```diff
@@ -0,0 +1,22 @@
+---
+id: adrs-adr013
+title: 'ADR013: Use node-fetch for data fetching'
+description: Architecture Decision Record (ADR) for HTTP data fetching packages
+---
+
+## Context
+
+Using multiple HTTP packages for data fetching increases the complexity and the
+support burden of keeping said package up to date.
+
+## Decision
+
+Node packages should use the `node-fetch` package for HTTP data fetching.
+Isomorphic packages should use the `cross-fetch` as a development dependency and
+rely on the built-in `fetch` API.
```

Comment:
I believe the wording should be that frontend code should `import fetch from 'cross-fetch';` for mockability purposes, but it's fine to use builtin types for `Request`, `RequestInit`, `Response` etc.

Controversial suggestion though. Should we go so far as to say that we recommend using `useApi(fetchApiRef)` in frontend code? Similar to `Link` and `Button` in `core-components`, this is our shot at wiring up behaviors, analytics, hooks or whatnot going forward, and it also speaks to the mocking aspect, making it correspond to what we do for other mock-worthy things in general.

As a matter of fact, this ADR might also benefit from mentioning that isomorphic code should preferably be written so as to accept a `typeof fetch` from `cross-fetch` in its initialization code, so that both frontend and backend can choose to inject their fetch implementation of choice.

@backstage/maintainers thoughts?